test/jdk/javax/xml/jaxp/datatype/8033980/SerializationTest.java

```diff
@@ -25,160 +25,314 @@
+     * second element contain object reference to GregorianCalendarAndDurationSerData specific to JDK version
+     * and third element contain expected Duration as string.
+     */
+    @DataProvider(name = "DurationData")
+    public Object [][] DurationData() {
+        return new Object [][] {{System.getProperty("java.version"), gregorianCalendarAndDurationSerData[0], EXPECTED_DURATION},
+                {JDK[0], gregorianCalendarAndDurationSerData[1], EXPECTED_DURATION},
+                {JDK[1], gregorianCalendarAndDurationSerData[2], EXPECTED_DURATION},
+                {JDK[2], gregorianCalendarAndDurationSerData[3], EXPECTED_DURATION},
+                {JDK[3], gregorianCalendarAndDurationSerData[4], EXPECTED_DURATION}};
+    }
+
+    /**
+     * Provide data for JDK version and Duration Base64 encode serialized bytes string.
+     * @return A two-dimensional Array of objects where each element is an array of size three. First element contain JDK version,
+     * second element contain object reference to GregorianCalendarAndDurationSerData specific to JDK version
+     * and third element contain expected Duration as string.
+     */
+    @DataProvider(name = "DurationDataBase64")
+    public Object [][] DurationDataBase64() {
+        return new Object [][] {{System.getProperty("java.version"), gregorianCalendarAndDurationSerData[0], EXPECTED_DURATION},
+                {JDK[2], gregorianCalendarAndDurationSerData[3], EXPECTED_DURATION},
+                {JDK[3], gregorianCalendarAndDurationSerData[4], EXPECTED_DURATION}};
     }
 
     /**
-     * Read the object from a file.
+     * Verify that GregorianCalendar serialized with different old JDK versions can be deserialized correctly with
+     * JDK under test.
+     * @param javaVersion JDK version used to GregorianCalendar serialization.
+     * @param gcsd JDK version specific GregorianCalendarAndDurationSerData.
+     * @param gregorianDate String representation of GregorianCalendar Date.
+     * @throws IOException Unexpected.
+     * @throws ClassNotFoundException Unexpected.
      */
-    private static Object fromFile(String filePath) throws IOException,
+    @Test(dataProvider="GregorianCalendarData")
+    public void testReadCalBytes(String javaVersion, GregorianCalendarAndDurationSerData gcsd, String gregorianDate) throws IOException,
             ClassNotFoundException {
-        InputStream streamIn = SerializationTest.class.getResourceAsStream(
-            filePath);
-        ObjectInputStream objectinputstream = new ObjectInputStream(streamIn);
-        Object o = objectinputstream.readObject();
-        return o;
+        final ByteArrayInputStream bais = new ByteArrayInputStream(gcsd.getGregorianCalendarByteArray());
+        final ObjectInputStream ois = new ObjectInputStream(bais);
+        final XMLGregorianCalendar xgc = (XMLGregorianCalendar) ois.readObject();
+        Assert.assertEquals(xgc.toString(), gregorianDate);
```

Comment:
if you:

`import static org.testng.Assert.*;`. then you can simply call `assertEquals`